src/Storages/MergeTree/MutateTask.cpp

```diff
@@ -1029,6 +1049,42 @@ void finalizeMutatedPart(
     new_data_part->default_codec = codec;
 }
 
+/// For alter table modify columns, check which option is set and which column is changed.
+static void getSecondaryIndicesOnColumnAlterModifyOptions(
+    MutationCommandsConstPtr commands,
+    MergeTreeSettingsPtr settings,
+    NameSet & alter_column_names,
+    bool & secondary_indices_on_columns_alter_modify_drop,
+    bool & secondary_indices_on_columns_alter_modify_rebuild)
+{
+    bool secondary_indices_on_columns_alter = false;
+    if (commands->size() == 1 && commands->front().ast)
+    {
+        if (auto * alter_cmd = commands->front().ast->as<ASTAlterCommand>();
+            alter_cmd && alter_cmd->type==ASTAlterCommand::MODIFY_COLUMN)
+        {
+            if (auto * column_declaration = alter_cmd->col_decl->as<ASTColumnDeclaration>())
+            {
+                alter_column_names.insert(column_declaration->name);
+                secondary_indices_on_columns_alter = true;
```

Comment:
I don't understand why we need variable `secondary_indices_on_columns_alter` if we could as well check in l. 1079 if `alter_column_names` is empty.